core/src/main/java/io/questdb/cairo/wal/WalEventWriter.java

```diff
@@ -104,6 +114,29 @@ private void appendBindVariableValuesByName(BindVariableService bindVariableServ
         }
     }
 
+    private int appendData(byte txnType, long startRowID, long endRowID, long minTimestamp, long maxTimestamp, boolean outOfOrder, long lastRefreshBaseTxn, long lastRefreshTimestamp) {
+        startOffset = eventMem.getAppendOffset() - Integer.BYTES;
+        eventMem.putLong(txn);
+        eventMem.putByte(txnType);
+        eventMem.putLong(startRowID);
+        eventMem.putLong(endRowID);
+        eventMem.putLong(minTimestamp);
+        eventMem.putLong(maxTimestamp);
+        eventMem.putBool(outOfOrder);
+        if (txnType == WalTxnType.MAT_VIEW_DATA) {
+            assert lastRefreshBaseTxn != Numbers.LONG_NULL;
```

Comment:
This must mark WAL segment as version 1 instead of 0 in WalUtils.WAL_FORMAT_OFFSET_32 offset when this happens.

Reply:
fixed in: https://github.com/questdb/questdb/pull/5507/commits/18dd32480f52eef2f3aac2b537db27d64cbf1cee